example/ExampleMain.cpp

```diff
@@ -132,6 +139,75 @@ std::unique_ptr<Module> createModuleExampleTypedPtrs(LLVMContext &context) {
   return module;
 }
 
+struct VisitorInnermost {
+  int counter = 0;
+};
+
+struct VisitorNest {
+  raw_ostream *out = nullptr;
+  VisitorInnermost inner;
+
+  void visitBinaryOperator(BinaryOperator &inst) {
+    *out << "visiting BinaryOperator: " << inst << '\n';
+  }
+};
+
+struct VisitorContainer {
+  int padding;
+  VisitorNest nest;
+};
+
+template <>
+struct llvm_dialects::VisitorPayloadProjection<VisitorNest, raw_ostream> {
+  static raw_ostream &project(VisitorNest &nest) { return *nest.out; }
+};
+
+LLVM_DIALECTS_VISITOR_PAYLOAD_PROJECT_FIELD(VisitorContainer, nest)
+LLVM_DIALECTS_VISITOR_PAYLOAD_PROJECT_FIELD(VisitorNest, inner)
+
+void exampleVisit(Module &module) {
+  VisitorBuilder<VisitorContainer> builder;
+
+  builder.nest<VisitorNest>(
+      [](VisitorBuilder<VisitorNest> &b) {
+        b.add<xd::ReadOp>([](VisitorNest &self, xd::ReadOp &op) {
+          *self.out << "visiting ReadOp: " << op << '\n';
+        });
+        b.add<UnaryInstruction>([](VisitorNest &self, UnaryInstruction &inst) {
+          *self.out << "visiting UnaryInstruction: " << inst << '\n';
+        });
+        b.add(&VisitorNest::visitBinaryOperator);
+        b.nest<raw_ostream>([](VisitorBuilder<raw_ostream> &b) {
```

Comment:
Maybe rename the builders to overload `b`, e.g. `nestedBuilder` and `innerStreamBuilder`?

Reply:
Hmm, I kind of like it that way :)